client/src/components/menu-views/new-menu-item-dish.js

```diff
@@ -0,0 +1,313 @@
+import { withRouter } from 'react-router-dom';
+import Api from '../../services/api';
+import View from '../shared/dropdown/view';
+import AlertBase from '../shared/alert-base';
+
+class NewMenuItemDish extends Component {
+  // eslint-disable no-param-reassign
+  constructor(props) {
+    super(props);
+    this.state = {
+      categories: [],
+      selectedCategory: 'Pizza',
+      selectedCategoryId: 1,
+      units: ['pcs', 'gram', 'L', 'ml'],
+      selectedPortionSize: 'pcs',
+      portionSize: 1,
+      portionPrice: 1,
+      dish: { categoryFood: { id: 0 } },
+      image: null,
+      isShowAlert: false,
+      errors: { err: 'Please fill all fields! ' },
+    };
+    this.fileInputRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this.getCategories('category');
+  }
+
+  onFileSelect(e) {
+    const file = e.target.files[0];
+    const reader = new FileReader();
+    const imageTag = document.getElementById('dishImage');
+
+    reader.onload = function (event) {
+      imageTag.src = event.target.result;
+    };
+    reader.readAsDataURL(file);
+    this.setState({ image: file });
+  }
+
+  onSelectClick() {
+    this.fileInputRef.current.click();
+  }
+
+  onCancelClick() {
+    this.props.selectedTab('menu');
+    this.props.history.goBack();
+  }
+
+  onAddClick() {
+    const {
+      dish, dishName, ingredients, selectedCategoryId, errors,
+    } = this.state;
+
+    if (!this.validateForm(errors)) {
+      this.setAlertState(true);
+    } else {
+      dish.name = dishName;
+      dish.ingredients = ingredients;
+      dish.categoryFood.id = selectedCategoryId;
+
+      this.sendData(dish);
+    }
+  }
+
+  getCategories(path) {
+    Api.get(path)
+      .then((response) => {
+        if (response.error == null) {
+          this.setState({
+            categories: response.data.content,
+          });
+        }
+      });
+  }
+
+  setAlertState(showAlert) {
+    this.setState({
+      isShowAlert: showAlert,
+    });
+  }
+
+  validateForm(errors) {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => { if (val.length > 0) valid = false; },
+    );
+    return valid;
+  }
+
+  handleChange(e) {
+    const { name, value } = e.target;
+    const { errors, dishName, ingredients } = this.state;
+    errors.err = '';
+    switch (name) {
+      case 'ingredients':
+        errors.ingredients = (value.length > 2 && value.length < 256)
+          ? '' : 'Ingredients field must be 3-255 characters long! ';
+        errors.err = this.checkIfEmptyFields(dishName, value);
+        break;
+      case 'dishName':
+        errors.name = (value.length > 0 && value.length < 31) ? '' : 'Dish name must be 1-30 characters! ';
+        errors.err = this.checkIfEmptyFields(value, ingredients);
+        break;
+      case 'portionPrice':
+        errors.price = value > 0 ? '' : 'The price cannot be negative or zero! ';
+        errors.price += value < 10000000 ? '' : 'The Portion price is too high! ';
```

Comment:
Lol)

Reply:
Nice) It works!

![image](https://user-images.githubusercontent.com/6152066/81062584-e4c1a000-8ede-11ea-8236-836b0249b0fd.png)